test/integration/backward_compatible/parallel/security/SimpleCredentials.js

```diff
@@ -24,13 +25,25 @@ class SimpleCredentials {
     }
 
     readData(input) {
-        this.username = input.readString();
-        this.password = input.readString();
+        // readString is added in 4.2 https://github.com/hazelcast/hazelcast-nodejs-client/pull/802
+        if (TestUtil.isClientVersionAtLeast('4.2')) {
+            this.username = input.readString();
```

Comment:
Minor: We could also make use of `TestUtil.readStringFromInput`. Same for the write part

Reply:
ok simpler, done in https://github.com/hazelcast/hazelcast-nodejs-client/pull/1170/commits/4ac8060aa87dbc1c366e9370ba2534eeea02b2d8